src/test/java/com/es/phoneshop/web/ProductListPageServletTest.java

```diff
@@ -24,18 +25,26 @@
     private HttpServletResponse response;
     @Mock
     private RequestDispatcher requestDispatcher;
+    @Mock
+    private ServletConfig servletConfig;
 
-    private ProductListPageServlet servlet = new ProductListPageServlet();
+    private final ProductListPageServlet servlet = new ProductListPageServlet();
 
     @Before
-    public void setup(){
+    public void setup() throws ServletException {
+        servlet.init(servletConfig);
         when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
     }
 
     @Test
     public void testDoGet() throws ServletException, IOException {
         servlet.doGet(request, response);
-
         verify(requestDispatcher).forward(request, response);
     }
+
+    @Test
+    public void testDaoFindProducts() throws ServletException, IOException {
+        servlet.doGet(request, response);
+        verify(request).setAttribute(eq("products"), any());
```

Comment:
Try to mock ProductDao and inject it into tested object with mockito, to verify that "products" attribute populated with what returned from productDao.findProduct(). You might need to update servlet to implement injection.